test/core/derivatives/problem.py

```diff
@@ -152,7 +152,8 @@ def forward_pass(
             input = subsample(input, dim=batch_axis_in, subsampling=subsampling)
 
         if input_requires_grad:
-            input.requires_grad = True
+            if input.dtype is float32:
+                input.requires_grad = True
```

Comment:
`RuntimeError: only Tensors of floating point and complex dtype can require gradients` is the error message if you try to assign `input.requires_grad=True` on long.
So yes, your suggestion would work fine. 
Do you think this is better? If yes, I'll change it.

Reply:
I would prefer it because it's less strict. We don't use complex tensors in BackPACK at the moment, but if we did, your code will not set `requires_grad`. At the same time, making the check more complicated by also checking whether `dtype` is complex makes the code unnecessarily complicated, because there are no complex tensors in BackPACK at the moment.

Reply:
Changed condition to `if input.dtype is not long`